cns/api.go

```diff
@@ -51,16 +53,70 @@ type HTTPService interface {
 // This is used for KubernetesCRD orchestrator Type where NC has multiple ips.
 // This struct captures the state for SecondaryIPs associated to a given NC
 type IPConfigurationStatus struct {
-	NCID      string
-	ID        string // uuid
-	IPAddress string
-	State     types.IPState
-	PodInfo   PodInfo
+	ID                   string // uuid
+	IPAddress            string
+	LastStateTransition  time.Time
+	NCID                 string
+	PodInfo              PodInfo
+	state                types.IPState
+	stateMiddlewareFuncs []stateMiddlewareFunc
+}
+
+// Equals compares a subset of the IPConfigurationStatus fields since a direct
+// DeepEquals or otherwise complete comparison of two IPConfigurationStatus objects
+// compares internal state details that don't impact their functional equality.
+//nolint:gocritic // it's safer to pass this by value
+func (i *IPConfigurationStatus) Equals(o IPConfigurationStatus) bool {
+	if i.PodInfo != nil && o.PodInfo != nil {
+		if !i.PodInfo.Equals(o.PodInfo) {
+			return false
+		}
+	}
+	return i.ID == o.ID &&
+		i.IPAddress == o.IPAddress &&
+		i.NCID == o.NCID &&
+		i.state == o.state
+}
+
+func (i *IPConfigurationStatus) GetState() types.IPState {
```

Comment:
Do we need a getter? Can we just make `State` exported?

Reply:
Oh wait I see, by making `state` unexported you are ~encouraging~ _forcing_ a state update to flow through `SetState` where you have the transition time recording. This is good, ignore my comment above (chose not to edit in case I look at this later and don't remember)